app/controllers/categorisation/CyaCategorisationController.scala

```diff
@@ -156,43 +155,49 @@ class CyaCategorisationController @Inject() (
 
   def onSubmit(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      CategoryRecord.build(request.userAnswers, request.eori, recordId, categorisationService) match {
-        case Right(categoryRecord) =>
-          auditService.auditFinishCategorisation(
+      goodsRecordConnector.getRecord(request.eori, recordId).flatMap { oldRecord =>
+        CategoryRecord
+          .build(
+            request.userAnswers,
             request.eori,
-            request.affinityGroup,
             recordId,
-            categoryRecord
-          )
-
-          val result = for {
-            oldRecord <- goodsRecordConnector.getRecord(request.eori, recordId)
-            _         <-
-              goodsRecordConnector
-                .updateCategoryAndComcodeForGoodsRecord(request.eori, recordId, categoryRecord, oldRecord)
-          } yield {
-            dataCleansingService.deleteMongoData(request.userAnswers.id, CategorisationJourney)
-            Redirect(
-              navigator.nextPage(
-                CyaCategorisationPage(recordId),
-                NormalMode,
-                request.userAnswers
-              )
+            categorisationService
+          ) match {
+          case Right(categoryRecord) =>
+            auditService.auditFinishCategorisation(
+              request.eori,
+              request.affinityGroup,
+              recordId,
+              categoryRecord
             )
-          }
 
-          result
+            val result = for {
+              _ <-
+                goodsRecordConnector
+                  .updateCategoryAndComcodeForGoodsRecord(request.eori, recordId, categoryRecord, oldRecord)
+            } yield {
+              dataCleansingService.deleteMongoData(request.userAnswers.id, CategorisationJourney)
+              Redirect(
+                navigator.nextPage(
+                  CyaCategorisationPage(recordId),
+                  NormalMode,
+                  request.userAnswers
+                )
+              )
+            }
 
-        case Left(errors) =>
-          dataCleansingService.deleteMongoData(request.userAnswers.id, CategorisationJourney)
-          Future.successful(
-            logErrorsAndContinue(
-              errorMessage,
-              controllers.categorisation.routes.CategorisationPreparationController.startCategorisation(recordId),
-              errors
+            result
```

Comment:
Unimportant (and existing code), but we can actually just return the for comp here instead of assigning it to a value and returning the value